File: scala/src/main/scala/com/laamella/code_state_machine/condition/AfterCondition.scala
package com.laamella.code_state_machine.condition;

/**
 * This condition is met after a certain amount of milliseconds.
 */
class AfterCondition[E](milliseconds: Long) extends NonEventBasedCondition[E] {
	private var minimalMeetTime: Long = _

	override def isMet = System.currentTimeMillis() > minimalMeetTime

	override def reset() = minimalMeetTime = System.currentTimeMillis() + milliseconds
}

File: ceylon/src/main/java/com/laamella/code_state_machine/condition/AfterCondition.java
package com.laamella.code_state_machine.condition;

/**
 * This condition is met after a certain amount of milliseconds.
 */
public final class AfterCondition<E> extends NonEventBasedCondition<E> {
	private final long milliseconds;
	private long minimalMeetTime;

	public AfterCondition(final long milliseconds) {
		this.milliseconds = milliseconds;
	}

	@Override
	public boolean isMet() {
		return System.currentTimeMillis() > minimalMeetTime;
	}

	@Override
	public void reset() {
		minimalMeetTime = System.currentTimeMillis() + milliseconds;
	}
}